### ISSUE TITLE
select.poll() is not supported on windows

### ISSUE THREAD

**Issue by dauheeIRL** [jmccrohan/pysolarmanv5#37]
I get the following error from windows 10 22H2, python 3.10.2, when _socket_setup is called in pysolarmanv5.py:
`AttributeError: module 'select' has no attribute 'poll'`

select.poll() is not supported on windows and since the commit below, compatibility is broken:

https://github.com/jmccrohan/pysolarmanv5/commit/bba4543bbaa469d54f43b76d2f8ef8d7d5889dff

Just wondering if you are interested in keeping compatible with windows and I could see if there is a workaround (with my crude abilities), or maybe you would be better suited.

**Comment by githubDante** [comment]
Sorry about that. I didn't mean to break anything. Already doing a research how to fix it.

**Comment by githubDante** [comment]
@dauheeIRL are you able to test [this branch](https://github.com/githubDante/pysolarmanv5/tree/win_compat) to confirm that it is OK now on Windows.

The tests are passing on Windows 11 with Python 3.11, but I don't have access to a datalogger from that machine to confirm personally.

**Comment by dauheeIRL** [comment]
Oh wow that was quick. What I've done is a loop to try every 20 seconds and it seems to generate an error periodically:
```
Exception in thread Thread-36 (_data_receiver):
Traceback (most recent call last):
  File "C:\Program Files\Python310\lib\threading.py", line 1009, in _bootstrap_inner
    self.run()
  File "C:\Program Files\Python310\lib\threading.py", line 946, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Program Files\Python310\lib\site-packages\pysolarmanv5\pysolarmanv5.py", line 298, in _data_receiver
    data = self.sock.recv(1024)
ConnectionAbortedError: [WinError 10053] An established connection was aborted by the software in your host machine
```


it also leaves a buildup of threads (QueueFeederThread + Process_Request_Thread), some of them seem to clear but others don't. when I call modbus.disconnect() after getting my logger data I get the following error:
```
Exception in thread Thread-41 (_data_receiver):
Traceback (most recent call last):
  File "C:\Program Files\Python310\lib\threading.py", line 1009, in _bootstrap_inner
    self.run()
  File "C:\Program Files\Python310\lib\threading.py", line 946, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Program Files\Python310\lib\site-packages\pysolarmanv5\pysolarmanv5.py", line 298, in _data_receiver
    data = self.sock.recv(1024)
OSError: [WinError 10038] An operation was attempted on something that is not a socket
```

I will have time tomorrow to do some extra testing

**Comment by githubDante** [comment]
Was able to reproduce this. If  the exit event is set early, however,  as in 638ed8e,  the socket is closed properly on disconnect.  Can you confirm this .

**Comment by dauheeIRL** [comment]
Thats working perfectly!! I hammered it every second did lots of connecting and disconnecting and couldn't get it to complain :)

Nice one, much appreciated

**Comment by githubDante** [comment]
Thank you for the cooperation!
By the way you can use the auto_reconnect=True option and to read/write indefinitely. There is no need to disconnect and create new instances all the time.

**Comment by dauheeIRL** [comment]
Thats great thanks. Yeah auto_reconnect=True has been working flawlessly

**Comment by jmccrohan** [comment]
Resolved by #38.

Thanks @dauheeIRL for the report, and @githubDante for the fix!
